### PR TITLE
Delete post

### LINKED ISSUE #9: PATCH/PUT Endpoint - Edit a project or palette title value

### LINKED ISSUE #11: DELETE Endpoint - Single Palette
api/v1/projects/:id/palettes/:id

### PR DESCRIPTION
**Pull request by Dunn-Austen**
### What's this PR do?
- DELETE (palette): Adds implementation code plus happy & sad path tests 
- PATCH (project title): Adds implementation code plus happy & sad path tests 
- Polishes the code for previously submitted endpoints & tests (emphasizing FE developer-friendly fetch responses)

### Any background information?
- Please heavily scrutinize the code for the PATCH endpoint and accompanying happy path test. Refer to http://knexjs.org/ for information on the update() method I employed from Knex.
- Documentation for the submitted endpoints is pending

### Pertinent Issues
- Partially addresses #18
- Partially addresses #24
- Resolves #11
- Resolves #9 

### Screenshots
- The current state of our test suite:

image

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -15,10 +14,17 @@ app.get('/', (req, res) => {
...
     const projects = await database('projects').select();
-    response.status(200).json(projects)
+    const cleanedProjects = projects.map(project => {
+      return {
+        id: project.id,
+        title: project.title
+      }
+    });
+
+    response.status(200).json({projects: cleanedProjects})
```
Comment (Dustin-Mikusko) [inline app.js]: I'm a huge fan of this format.

File: app.test.js [hunk 2]
```diff
@@ -22,14 +22,13 @@ describe('Server', () => {
...
       const expectedProject = await database('projects').first();
+      const { id } = expectedProject;
         delete expectedProject.created_at;
         delete expectedProject.updated_at;
-      const { id } = expectedProject;
 
       const response = await request(app).get(`/api/v1/projects/${id}`);
-      const result = response.body[0];
-        delete result.created_at;
-        delete result.updated_at;
```
Comment (Dustin-Mikusko) [inline app.test.js]: I'm going to miss this. It looked very clean, but I do like the new way

File: app.test.js [hunk 3]
```diff
@@ -79,24 +74,70 @@ describe('Server', () => {
...
+  describe('PATCH /api/v1/projects/:id', () => {
+    it('should return a happy status of 201 and update a specific project title', async () => {
+      const newProject = { title: 'Master bedroom' };
+      const targetProject = await database('projects').first();
+      const { id } = targetProject;
+
+      const response = await request(app).patch(`/api/v1/projects/${id}`).send(newProject)
+
+      const revisedProjectArray = await database('projects').where('id', id);
+      const revisedProject = revisedProjectArray[0];
```
Comment (Dustin-Mikusko) [inline app.test.js]: This here is the whole object coming back with the changed properties? If so, that's perfect!

File: app.test.js [hunk 4]
```diff
@@ -79,24 +74,70 @@ describe('Server', () => {
...
+        delete revisedProject.created_at;
+        delete revisedProject.updated_at;
+
+      expect(response.status).toBe(201);
+      expect(revisedProject).toEqual(response.body);
```
Comment (Dustin-Mikusko) [inline app.test.js]: Maybe have a mock object to look the way it's expected rather than `response.body` to convey what we expect to be returned

Comment (Dunn-Austen) [reply app.test.js]: Will do. I'll make the edit and get the branch pushed back up

File: app.test.js (hunk 3 above)
Comment (Dunn-Austen) [reply app.test.js]: Yep {id: _, title: _}

File: app.test.js [hunk 5]
```diff
@@ -79,24 +74,62 @@ describe('Server', () => {
...
+
+      const revisedProjectArray = await database('projects').where('id', id);
+      const revisedProject = revisedProjectArray[0];
+        delete revisedProject.created_at;
+        delete revisedProject.updated_at;
+
+      const expectedResult = {id, title: 'Master bedroom' };
+
+      expect(response.status).toBe(201);
+      expect(revisedProject).toEqual(expectedResult);
```
Comment (Dustin-Mikusko) [inline app.test.js]: PERFECT

**Pull request merged**